bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/discovery/WemoDiscoveryParticipant.java

```diff
@@ -63,6 +64,8 @@
                 // ignore and use default label
             }
             properties.put(UDN, device.getIdentity().getUdn().getIdentifierString());
+            URL descriptorURL = device.getIdentity().getDescriptorURL();
```

Comment:
Can it be `null` here also?

Reply:
No, it cannot be null at this point. If the device does not deliver a proper descriptorURL at discovery, the device will not be „fully hydrated“ and put in the UPnP registry. Therefore it would not be discovered by openHAB.